OpenTrons/opentrons-api
# feat(protocol-designer): foundation for waste chute entity in redux

closes RAUT-682, RAUT-686, RAUT-724, RAUT-727, RAUT-728, RAUT-748

# Overview

This create a concept of a waste chute entity in PD by setting up the ground work for adding "deck fixture" entities via the Redux state and creating `AdditionalEquipmentEntities` to be used in active deck setup

Flex staging areas will be able to use the same redux actions and will be included as an additional equipment entity.

# Test Plan

sandbox: https://sandbox.designer.opentrons.com/pd_add-waste-chute-entity/

- Go to settings and turn on Enable Deck modification feature flag (might need to turn your dev ffs on if its not there by typing in `enablePrerelaseMode()` in console and refreshing)
- Create a flex protocol, on the modules and other tab, select the Waste Chute (i am using the gripper as a placeholder asset rn). Click on the "review file details" button to create the protocol
- open up dev tools and go to the redux tab. Go to `StepForms` -> `additionalEquipmentInvariantProperties`. Open that up and see the waste chute entity. Should include name, id, and location as D3. 
- scroll down in PD and find the Additional Items section. Try removing and adding the waste chute, see that the redux state properly adds/removes the waste chute entity
- add a module, if you select to add it to slot `D3`, see that there is an error saying the slot is full 
- Now, create a new protocol for the OT-2, see that the waste chute isn't selectable
- Now, turn off the Enable Deck modification ff, create a new protocol for the Flex, see that the Waste Chute is not selectable.
- update gripper asset

# Changelog

- extended the `CreateFileWizard`'s `ModulesAndOtherTile` to include wasteChute as a value in formik and also to make the wasteChute button
- created new redux types: `CREATE_DECK_FIXTURE` and `DELETE_DECK_FIXTURE` and create actions and add them to reducers -> these same redux actions will be used for flex staging
- create type `additionalEquipmentEntities` and util `getAdditionalEquipmentEntities` from grabbing the `additionalEquipmentInvariantProperties` and add to `_getInitialDeckSetup` so the deck setup has knowledge of the waste chute
- change `GripperRow` to `AdditionalItemsRow` to have both waste chute and gripper and create a test

# Review requests

All the visual stuff is hidden behind the ff. Theoretically, even with the changes added to`_getInitialDeckSetup`, it should be fine since you can't create a waste chute entity to begin with. I think ideally, I don't want this PR in PD 7.0 release BUT since the visuals are hidden behind a FF, it might be fine? 

# Risk assessment

low
